Task2/Program.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Task2;
+
+public class Program
+{
+    public static (string, int) Bwt(string str)
+    {
+        var result = "";
+        var suffixArray = GetSuffixIndices(str + str);
+        suffixArray = suffixArray.Where(x => x < str.Length).ToArray();
+        foreach (var index in suffixArray)
+        {
+            if (index == 0)
+            {
+                result += str.Last();
+                continue;
+            }
+
+            result += str[index - 1];
+        }
+
+        return (result, Array.IndexOf(suffixArray, 0));
+    }
+
+    public static string ReverseBwt(string str, int initialPosition)
+    {
+        var result = "";
+        var alphabet = new List<char>();
+        foreach (var el in str)
+        {
+            if (!alphabet.Contains(el))
+            {
+                alphabet.Add(el);
+            }
+        }
```

Comment:
```suggestion
        var alphabet = new SortedSet<char>(str).ToList();
```
И быстрее (n logn вместо n^2), и короче.
(-0.1)